--- /Users/matthias/SDKs/JUCE/modules/juce_gui_basics/widgets/juce_Slider.h	2014-01-07 09:11:53.000000000 +0200
+++ JUCE/modules/juce_gui_basics/widgets/juce_Slider.h	2014-01-09 14:07:01.000000000 +0200
@@ -615,6 +615,12 @@
     */
     void setScrollWheelEnabled (bool enabled);
 
+    /** Set the slider Wheel to endless encoder type.
+     
+     By default it's disabled.
+     */
+    void setScrollWheelEndless (bool enabled);
+    
     /** Returns a number to indicate which thumb is currently being dragged by the mouse.
 
         This will return 0 for the main thumb, 1 for the minimum-value thumb, 2 for
